verifiers/verifier.go

```diff
@@ -37,14 +37,15 @@ func getVerifier(builderOpts *options.BuilderOpts) (register.SLSAVerifier, error
 
 func VerifyImage(ctx context.Context, artifactImage string,
```

Comment:
Some folks use slsa-verifier's Go API. Could we maybe add a new method `VerifyImageProvenanceRepo` with the new signature rather than modify `VerifyImage` in order to maintain API backwards compatibility?

Or maybe it makes sense to add another `options.ImageOptions` (with just `ProvenanceRepository` on it for now) and add a `VerifyImageWithOptions` function instead in order to protect against future API changes? WDUT?

/cc @laurentsimon